flask_saml2/sp/sp.py

```diff
@@ -306,22 +328,35 @@ def get_metadata_context(self) -> dict:
             'contacts': [],
         }
 
-    def create_blueprint(self) -> Blueprint:
+    def get_scheme(self) -> str:
+        return self.scheme
+
+    def get_acs_redirect_endpoint(self) -> str:
+        return self.acs_redirect_endpoint
+
+    # With acs_redirect_url, you can set the url that the Access Consumer Service redirects to upon successful login
+    # This is unnecessary if you expect a "relay_state" parameter in the SAML request to the ACS
+    def create_blueprint(self, login_endpoint='/login/', login_idp_endpoint='/login/idp/',
+                         logout_endpoint='/logout/', acs_endpoint='/acs/', sls_endpoint='/sls/',
+                         metadata_endpoint='/metadata.xml', scheme='http') -> Blueprint:
```

Comment:
I’m far from sure about this, but shouldn’t the new `entity_id` not also be passed as an argument here, same as the scheme? (And then of course also something at line 345)

```suggestion
    def create_blueprint(self, login_endpoint='/login/', login_idp_endpoint='/login/idp/',
                         logout_endpoint='/logout/', acs_endpoint='/acs/', sls_endpoint='/sls/',
                         metadata_endpoint='/metadata.xml', scheme='http', entity_id=None) -> Blueprint:
```

Reply:
I thought about that. The main reason I didn't pass `entity_id` there is that it seemed to be a little outside the scope of what this `create_blueprint` function should do. To me, it makes sense to set the `scheme` together with the various endpoints, as they are related. But you could make a case that even `scheme` doesn't belong here, as it is setting a value in `sp` rather than in `idp_bp`. 

I'm open to including `entity_id` or removing `scheme`. Maybe creating a separate `config_and_create_blueprint` function which sets the `entity_id` and `scheme` and then calls `create_blueprint`. Or other ideas? Not sure what the cleanest solution is.

Reply:
Aah, yeah, got it. In that case I’m out of my depth, I wouldn’t know what the cleanest solution would be I’m afraid.